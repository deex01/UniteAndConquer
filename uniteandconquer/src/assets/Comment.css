.comment {
  display: grid;
  grid-template-columns: 30px 0.8fr;
  background: #ffd9a0;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 5px;
  padding: 5px;
  margin-top: 15px;
  width: 350px;
}

.comment-name {
  font-weight: bold;
}

.comments {
  border: 1px solid black;
  background: rgb(255, 229, 193);
  padding: 25px;
  border-radius: 5px;
  width: 70%;
}

.add-comment {
  margin-top: 30px;
  display: grid;
  grid-template-columns: 30px 0.9fr 0.1fr;
}

.input-comment {
  height: 10px;
  font-size: 15px;
  background: #ffd9a0;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 5px;
}

.submit-comment {
  border: none;
  background: #FFCB7D;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 5px;
  padding: 15px;
  height: 30px;
  font-style: italic;
  font-weight: bold;
}

.button-text {
  position: relative;
  bottom: 6px;
}

.far.fa-user-circle {
  position: relative;
  top: 8px;
  left: 5px;
}