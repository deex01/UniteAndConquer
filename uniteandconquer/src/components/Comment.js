import React, { useState } from 'react';
import '../assets/Comment.css';

function Comment() {
  const [tempID, setTempID] = useState(3);
  const [commentInput, setCommentInput] = useState('');
  const [comments, setComments] = useState([
    { id: 1, name: 'user1', content: 'I loved using this item' },
    { id: 2, name: 'user2', content: 'I wonder if I will need this' },
  ]);

  const addComment = () => {
    const newComment = { id: tempID, name: `user${tempID}`, content: commentInput };
    setComments([...comments, newComment]);
    setTempID(tempID + 1);
  };

  return (
    <div className="comments">
      {comments.map((comment) => (
        <div key={comment.name} className="comment">
          <div className="comment-img"><i className="far fa-user-circle " /></div>
          <div className="comment-text">
            <div className="comment-name">{comment.name}</div>
            <div className="comment-content">{comment.content}</div>
          </div>
        </div>
      ))}
      <div className="add-comment">
        <i className="far fa-user-circle" />
        <input data-testid="comment-input" className="input-comment" onChange={(e) => setCommentInput(e.target.value)} />
        <button className="submit-comment" type="button" onClick={addComment}><div className="button-text">Comment</div></button>
      </div>
    </div>
  );
}

export default Comment;
