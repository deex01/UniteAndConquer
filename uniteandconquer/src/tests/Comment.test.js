/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import Comment from '../components/Comment';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('add comment', () => {
  render(<Comment />);
  const comment = screen.getByTestId('comment-input');
  fireEvent.change(comment, { target: { value: 'Example comment text' } });
  expect(comment.value).toBe('Example comment text');
  fireEvent.click(screen.getByText('Comment'));
});

test('snapshot comments', () => {
  const component = renderer.create(<Comment />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
