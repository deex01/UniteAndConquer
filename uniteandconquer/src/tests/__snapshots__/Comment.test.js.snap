// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot comments 1`] = `
<div
  className="comments"
>
  <div
    className="comment"
  >
    <div
      className="comment-img"
    >
      <i
        className="far fa-user-circle "
      />
    </div>
    <div
      className="comment-text"
    >
      <div
        className="comment-name"
      >
        user1
      </div>
      <div
        className="comment-content"
      >
        I loved using this item
      </div>
    </div>
  </div>
  <div
    className="comment"
  >
    <div
      className="comment-img"
    >
      <i
        className="far fa-user-circle "
      />
    </div>
    <div
      className="comment-text"
    >
      <div
        className="comment-name"
      >
        user2
      </div>
      <div
        className="comment-content"
      >
        I wonder if I will need this
      </div>
    </div>
  </div>
  <div
    className="add-comment"
  >
    <i
      className="far fa-user-circle"
    />
    <input
      className="input-comment"
      data-testid="comment-input"
      onChange={[Function]}
    />
    <button
      className="submit-comment"
      onClick={[Function]}
      type="button"
    >
      <div
        className="button-text"
      >
        Comment
      </div>
    </button>
  </div>
</div>
`;
